### ISSUE TITLE
Update generators to use ideal template project

### ISSUE THREAD

**Issue by yeekangc** [OpenLiberty/open-liberty#9890]
Various starters or generators out there e.g. MP Start, RHR Launcher, IBM Cloud produce starter or template projects for developers to build with Open Liberty.  This feature is to review and ensure that these starters or generators are using the recommended (ideal) template project for Open Liberty.

- Define and build an ideal template project that developers can easily use to build their applications -- This will be based on what has been defined in #12373.
- Ensure that the template project is used consistently by downstream consumers and generators
- Include relevant considerations for common tasks like shared libraries and common dependencies -- Copy dependencies support will be provided by https://github.com/OpenLiberty/ci.maven/issues/705.
- Consider and optimize for development workflow and automated build/test/DevOps workflow -- To be covered by separate features.  See https://github.com/OpenLiberty/ci.maven/issues/739 and https://github.com/OpenLiberty/open-liberty/issues/14048.

@gkwan-ibm @scottkurz

**Comment by yeekangc** [comment]
FYI, @meswan.

**Comment by atosak** [comment]
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## List of Steps to complete or get approvals / sign-offs for Onboarding to the Liberty release (GM date)
 
Instructions:
 - Do the actions below and mark them complete in the checklist when they are done.
 - Make sure all feature readiness approvers put the appropriate tag on the epic to indicate their approval.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **Before Development Starts or 8 weeks before Onboarding** 
- [ ] POC Design / WAD Review Scheduled (David Chang) or N/A.
- [ ] POC Design / WAD Reviewed (Feature Owner) or N/A.
- [ ] Complete any follow-ons from the POC Review.
- [ ] Design / WAD Approval (Alasdair Nottingham) or N/A.
- [ ] No Design / No WAD Approval (Arthur De Magalhaes - cloud / Alasdair Nottingham - server) or N/A.
- [ ] SVT Requirements identified. (Epic owner / Feature owner with SVT focal point)
- [ ] ID Requirements identified. (Epic owner / Feature owner with ID focal point)
- [ ] Create a child task of the epic entitled "FAT Approval Test Summary".  Add and fill in the template as described here: https://github.ibm.com/was-liberty/WS-CD-Open/wiki/Feature-Review-(Feature-Test-Summary-Process)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **3 weeks before Onboarding** 
- [ ] Identify all open source libraries that are changing or are new. Work with Legal Release Services (Cass Tucker or Release PM) to get open source cleared and approved. Or N/A. (Epic Owner).   New or changed open source impacts license and Certificate of Originality.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE ** 3 weeks before Onboarding** 
- [ ] All new or changed PII messages are checked into the integration branch, before the last translation shipment out. (Epic Owner)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## TARGET COMPLETION DATE **2 weeks before Onboarding** 
- [ ] Implementation complete. (Epic owner / Feature owner) 
- [ ] All function tests complete. Ready for FAT Approval. (Epic owner / Feature owner)
- [ ] Review all known issues for Stop Ship. (Epic owner / Feature owner / PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## APPROVALS with TARGET COMPLETION DATE **2 to 1 week before Onboarding** 
Prereq: You must have the Design Approved or No Design Approved label on the GitHub Epic.
- [ ] **Accessibility** - (G Scott Johnston). Accessibility testing is complete or N/A. Approver adds label focalApproved:accessibility to the Epic in Github.
- [ ] **FAT Liberty SOE** - (Kevin Smith). SOE FATS are running successfully or N/A . Approver adds label focalApproved:fat to the Epic in Github.
- [ ] **Globalization** (Sam Wong - Liberty / Simy Cheeran - tWAS). Translation is complete or N/A. TVT - complete or N/A. Approver adds label focalApproved:globalization to the Epic in Github.
- [ ] **ID** - (Kareen Deen). Documentation work is complete or N/A . Approver adds label focalApproved:id to the Epic in Github.
- [ ] **Performance** - (Jared Anderson). Performance testing is complete with no high severity defects or N/A . Approver adds label focalApproved:performance to the Epic in Github.
- [ ] **Serviceability** - (Don Bourne). Serviceability has been addressed. 
- [ ] **STE** - (Swati Kasundra). STE chart deck is complete or N/A . Approver adds label focalApproved:ste to the Epic in Github.
- [ ] **SVT** - (Greg Ecock - Cloud, Brian Hanczaryk- APS). SVT is complete or N/A . Approver adds label focalApproved:svt to the Epic in Github.
- [ ] **Demo** - (Liberty only - Tom Evans or Chuck Bridgham). Demo is scheduled for an upcoming EOI. Approver adds label focalApproved:demo to the Epic in Github.
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## TARGET COMPLETION DATE **1 week before Onboarding** 
- [ ] No Stop Ship issues for the feature. (Epic owner / Feature owner / Release PM)
- [ ] Ship Readiness Review and Release Notes completed (Epic owner / Feature owner / Release PM)
- [ ] Github Epic and Epic's issues are closed / complete. All PRs are committed to the master branch. (Epic owner / Feature owner / Backlog Subtribe PM)
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
 ## NOT REQUIRED FOR A FEATURE
- [ ] **OL Guides** - (Yee-Kang Chang). Assessment for OL Guides is complete or N/A. 
- [ ] **WDT** - (Leonard Theivendra). WDT work complete or N/A.
 
 
- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
## Related Deliverables TARGET COMPLETION DATE **General Availability** 
- [ ] Blog article writeup (Epic owner / Feature owner / Laura Cowen)

**Comment by yeekangc** [comment]
Updated description with latest scope and pointers to related features.

**Comment by covener** [comment]
UFO Feedback/Actions:

- As a starter/generator developer ... reword to reflect who that persona is?
- upfront: clarify "the deliverable"
   - best practice published separately? This UFO (likely not enough)?
   - libertstarter + doc
- p10: Not necessarily latest versions, needs to be able to be specified.
   - what about jakarta EE9 transition?
- p10: latest liberty version?  via docker tag.
- p11: separate value from 'X' for will not fix, or draw line through row
   - don't update Codewind [further].
- p11: microshed test + OL.io not listed
- p11: eclipse starter is glassfish-only
- p13: health check added by generator needs to match mpHealth level?
- How will we keep known generators "compliant"?
- Test/IT info on how used? The targets?
- p18: don't encourage any default passwords [for keystores]
  - look into mpstarter behavior
- p25: don't mention CLI

**Comment by yeekangc** [comment]
Adding Design Approval Request for @gkwan-ibm.  @gkwan-ibm, can you please update with pointer to revised UFO?

**Comment by gkwan-ibm** [comment]
The revised UFO is at https://ibm.ent.box.com/file/734356892041

**Comment by NottyCode** [comment]
@gkwan-ibm @yeekangc you need to provide a publicly accessible UFO link

**Comment by gkwan-ibm** [comment]
The UFO is at https://ibm.box.com/s/4qtq82a6umnwmzqom3p9bpghcret1pup.

**Comment by gkwan-ibm** [comment]
@NottyCode I posted the public link in the above comment. Thanks

**Comment by gkwan-ibm** [comment]
- Fixed the following tools:
  - Bluemix CLI by https://github.com/IBM/java-liberty-app/pull/58
  - Codewind by https://github.com/codewind-resources/openLibertyTemplate/pull/6
  - RHR Launcher (PurpleLiberty) by
    - https://github.com/PurpleLiberty/example-health-check/pull/2
    - https://github.com/PurpleLiberty/example-cache/pull/1
    - https://github.com/PurpleLiberty/example-circuit-breaker/pull/2
    - https://github.com/PurpleLiberty/example-configmap/pull/2
    - https://github.com/PurpleLiberty/example-rest-http-secured/pull/2
    - https://github.com/PurpleLiberty/example-rest-http-crud/pull/2
    - https://github.com/PurpleLiberty/example-rest-http/pull/2
  - MP Starter is aware by https://github.com/eclipse/microprofile-starter/issues/339
  - Liberty Starter was acknowledged and fixed by https://github.com/OpenLiberty/start.openliberty.io/issues/2
- Created the template repository at https://github.com/OpenLiberty/project-templates

**Comment by gkwan-ibm** [comment]
All worked were done.

**Comment by gkwan-ibm** [comment]
> The revised UFO is at https://ibm.ent.box.com/file/734356892041

@NottyCode would you review/approve the UFO?

**Comment by NottyCode** [comment]
@gkwan-ibm Sorry for the delay here. I had an email discussion with some of our friends at Red Hat earlier in January. They observed the same thing we had that the RH Launcher support was not right, but in the course of that discussion we concluded that we shouldn't invest in fixing it, so I'd suggest changing the RH Launcher row to will not fix.

I've noticed that a few of the screenshots are out of date with regard to versions so I think it would be useful to have a separate table towards the end detailing the definitive dependencies and versions that will be added.
